File: webapp-net/Search/Data/SearchResult.cs
using System.Collections;

namespace Sdl.Web.Modules.Search.Data
{
    public class SearchResult
    {
        public string Id { get; set; }

        public string Content { get; set; }

        public string Language { get; set; } = "en";

        public string ProductFamilyTitle { get; set; }

        public string ProductReleaseVersionTitle { get; set; }

        public int PublicationId { get; set; }

        public string PublicationTitle { get; set; }

        public string PageId { get; set; }

        public string PageTitle { get; set; }

        public string LastModifiedDate { get; set; }

        public string ItemType => "Page";

        public IDictionary Meta { get; set; }

        public IDictionary Highlighted { get; set; }
    }
}

File: webapp-java/dxa-module-tridion-docs-search/src/main/java/com/sdl/dxa/modules/docs/search/model/SearchResult.java
package com.sdl.dxa.modules.docs.search.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import java.util.List;
import java.util.Locale;
import java.util.Map;

/**
 * Search Result represents data structure for single search result.
 */
@Data
@JsonIgnoreProperties(ignoreUnknown = true)
public class SearchResult {
    @JsonProperty("Id")
    private String id;

    @JsonProperty("Meta")
    private Map<String, Object> fields;

    @JsonProperty("Locale")
    private Locale locale;

    @JsonProperty("Highlighted")
    private Map<String, List<String>> highlighted;

    @JsonProperty("Content")
    private String content;

    @JsonProperty("CreatedDate")
    private String createdDate;

    @JsonProperty("ModifiedDate")
    private String modifiedDate;

    @JsonProperty("PublicationId")
    private Integer publicationId;

    @JsonProperty("PublicationTitle")
    private String publicationTitle;

    @JsonProperty("ProductFamilyName")
    private String productFamilyName;

    @JsonProperty("ProductReleaseName")
    private String productReleaseName;
}
